---
title: "BRAVE Kids 16S Analysis"
author: "Alex McCumber"
date: "11/3/2020"
output: html_document
---

```{r, include=FALSE}
#load some libraries
library(dada2)
library(dplyr)
library(tibble)
library(purrr)
library(DECIPHER)
library(phyloseq)
library(ggplot2)
library(Biostrings)
library(ggtree)
library(ape)
library(stringr)
library(vegan)
library(magrittr)
library(ape)
library(philr)
library(glmnet)
library(DESeq2)
library(tidyverse)
library(cluster)
library(devtools)
library(decontam)
library(picante)
library(vegan)
library(phyloseq)
library(tidyr)
library(nlme)
library(tidyverse)
library(ggplot2)
library(dplyr)
library(compositions)
source("~/BRAVE_Kids/ancom_v2.1.R")
```

```{r}
dir.create("~/BRAVE_Kids/")
```


#Set filepaths and get sample names
```{r, Set filepaths}
path = "/sharedspace/BraveKids/"

list.files(path)

fnFs = sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs = sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))

sample.names = sapply(strsplit(basename(fnFs),"_L001"), `[`,1)
```


```{r}
plotQualityProfile(fnFs[1:20],aggregate = TRUE)
```


```{r}
plotQualityProfile(fnRs[1:20], aggregate = TRUE)
```


```{r, set file path for where filtered reads will go}
filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
```


```{bash}
gunzip /sharedspace/BraveKids/PCOV1001_S1_L001_R1_001.fastq.gz

head -n 20 /sharedspace/BraveKids/PCOV1001_S1_L001_R1_001.fastq 

gzip /sharedspace/BraveKids/PCOV1001_S1_L001_R1_001.fastq

```


#Filter and trim reads, using standard calls and trimming 20 off each end to remove primers
```{r}
#set multithread to False if you're running Windows

out <- filterAndTrim(fnFs, filtFs, fnRs, filtRs, truncLen=c(240,220),
              maxN=0, maxEE=c(2,2), truncQ=2, rm.phix=TRUE,
              compress=TRUE, multithread=T)
head(out)

as.data.frame(out) %>%
  mutate(., sum(reads.out)/sum(reads.in))

#Remove samples with 0 reads
filter.out = out %>%
  as.data.frame(.) %>%
  rownames_to_column('samples') %>%
  filter(., reads.out > 1) %>%
  column_to_rownames('samples')

filter.samples = out %>%
  as.data.frame(.) %>%
  rownames_to_column('samples') %>%
  filter(., reads.out > 1) %>%
  column_to_rownames('samples') %>%
  rownames(.) 

#Re-establish sample names now that some are gone
sample.names = sapply(strsplit(basename(filter.samples),"_L001"), `[`,1)

filtFs <- file.path(path, "filtered", paste0(sample.names, "_F_filt.fastq.gz"))
filtRs <- file.path(path, "filtered", paste0(sample.names, "_R_filt.fastq.gz"))
```


#Generate error function, dereplicating, and merging reads
```{r, include=false}
#set multithread to False if you're running Windows
errF = learnErrors(filtFs, multithread = T)
errR = learnErrors(filtRs, multithread = T)

dadaFs <- dada(filtFs, err=errF, multithread=TRUE, pool = T)
dadaRs <- dada(filtRs, err=errR, multithread=TRUE, pool = T)

mergers <- mergePairs(dadaFs, filtFs, dadaRs, filtRs, verbose=TRUE)
```

```{r}
plotErrors(errF, nominalQ=TRUE)
```


```{r, making a sequence table}
seqtab = makeSequenceTable(mergers)

table(nchar(getSequences(seqtab)))

#Too many reads of lengths outside of anticipated amplicon size so will be removing all reads that don't fall within 250 to 260 bp
ncol(seqtab)

seqtab2=seqtab[,nchar(colnames(seqtab)) %in% seq(240,260)]

saveRDS(seqtab, "~/BRAVE_Kids/seqtab.RDS")

seqtab.nochim = removeBimeraDenovo(seqtab2, method="consensus", multithread = T, verbose = TRUE)

#3546 ASVs left
saveRDS(seqtab.nochim, "~/BRAVE_Kids/seqtab_nochim.RDS")

getN <- function(x) sum(getUniques(x))
track <- cbind(filter.out, sapply(mergers, getN), rowSums(seqtab.nochim))
colnames(track) <- c("input", "filtered", "merged", "nonchim")
rownames(track) <- sample.names
head(track)
colSums(track)
```

#Run this if you need to download the silva dataset v138 for IDTaxa function
```{bash, include = false}
wget http://www2.decipher.codes/Classification/TrainingSets/SILVA_SSU_r138_2019.RData -P ./silvadb
```


```{r}
dna <- DNAStringSet(getSequences(seqtab.nochim)) # Create a DNAStringSet from the ASVs

load("~/BRAVE_Kids/silvadb/SILVA_SSU_r138_2019.RData")

ids <- IdTaxa(dna, trainingSet, strand="top", processors=12, verbose=FALSE)

ranks <- c("domain", "phylum", "class", "order", "family", "genus") # ranks of interest

# Convert the output object of class "Taxa" to a matrix analogous to the output from assignTaxonomy
taxid <- t(sapply(ids, function(x) {
        m <- match(ranks, x$rank)
        taxa <- x$taxon[m]
        taxa[startsWith(taxa, "unclassified_")] <- NA
        taxa
}))
colnames(taxid) <- ranks; rownames(taxid) <- getSequences(seqtab.nochim)
```

##Creation of tree, need to run through python 2

```{bash, make a tree with SEPP using gg backbone tree}
wget  "https://raw.github.com/smirarab/sepp-refs/master/gg/sepp-package.tar.bz"

tar xvfj sepp-package.tar.bz

cd sepp-package/sepp

python setup.py config -c

python setup.py install --user
```

```{r, make list of sequences in fasta file for tree making}
uniques=getUniques(seqtab.nochim)

sequences=names(uniques)

uniquesToFasta(uniques,fout = "~/BRAVE_Kids/seqs.fasta", ids=sequences)
```

```{bash}
./sepp-package/run-sepp.sh seqs.fasta BRAVE
```

```{r, view tree file with ggtree}
seq.tree = read.tree(file = "~/BRAVE_Kids/BRAVE_placement.tog.relabelled.tre") %>%
  keep.tip(., sequences)
```


#Import metadata and make dataframe
```{r}
metad = read_csv("~/BRAVE_Kids/metadata_16sv2.csv")

rownames(metad) = str_remove_all(metad$brave_id, "-")

metaDF = subset(metad, rownames(metad) %in% gsub("_.*","",sample.names))

rownames(metaDF) = str_remove_all(metaDF$brave_id, "-")

metaDF$X1=NULL

rm(metad)
```

#Create phyloseq object
```{r}
#need to change seqtab.nochim to have same sample names as metadata
rownames(seqtab.nochim) = gsub("_.*","",rownames(seqtab.nochim))

ps = phyloseq(otu_table(seqtab.nochim, taxa_are_rows=FALSE), 
               sample_data(metaDF), 
               tax_table(taxid),
              phy_tree(seq.tree))

ps.Bact=subset_taxa(ps, domain=="Bacteria") 

saveRDS(ps.Bact, "~/BRAVE_Kids/ps.RDS")

rm(ps)
```

```{r}
subset_samples(ps, sample_data(ps)$brave_id == "extractNEGctrl") %>%
  prune_species(speciesSums(.) > 0, .)

#59 ASVs in extractNEGctrl

subset_samples(ps, sample_data(ps)$brave_id == "PCRneg") %>%
  prune_species(speciesSums(.) > 0, .)

#4 ASVs in PCRneg
```

#Decontam 
```{r}
library("decontam")
contamdf.freq = isContaminant(ps.Bact, method="frequency", conc = "dna", threshold = 0.1)

hist(contamdf.freq$p)

table(contamdf.freq$contaminant)

#Identified 40 contaminant taxa in at 0.1
#remove contaminant taxa

ps.noncontam = prune_taxa(!contamdf.freq$contaminant, ps.Bact)

set.seed(42)
plot_frequency(ps.Bact, taxa_names(ps.Bact)[sample(which(contamdf.freq$contaminant),4)], conc="dna") +
    xlab("DNA Concentration")
```

```{r}
subset_samples(ps.noncontam, sample_data(ps.noncontam)$brave_id == "extractNEGctrl") %>%
  prune_species(speciesSums(.) > 0, .)

#54 ASVs in extractNEGctrl

subset_samples(ps.noncontam, sample_data(ps.noncontam)$brave_id == "PCRneg") %>%
  prune_species(speciesSums(.) > 0, .)

#2 PCR Neg Ctrl ASVs 
```

#Remove contaminants ID'ed by decontam and remove ctrl and non-study samples
```{r, mesesage = FALSE}
ps.noncontam = prune_taxa(!contamdf.freq$contaminant, ps.Bact) %>%
  subset_samples(., sample_data(ps.noncontam)$sex != "NA") %>%
  prune_samples(sample_sums(.)>=1000, .) %>%
  #transform_sample_counts(.,function(x) x/sum(x)) %>%
  prune_species(speciesSums(.) > 0, .) 

ps.noncontam

saveRDS(ps.noncontam, "~/BRAVE_Kids/psBRAVE.RDS")

# Number ASVs and store the exact sequences in the reference sequence slot - can access with refseq(ps)
dna <- Biostrings::DNAStringSet(taxa_names(ps.noncontam))
names(dna) <- taxa_names(ps.noncontam)
phy.brave <- merge_phyloseq(ps.noncontam, dna)
taxa_names(ps.noncontam) <- paste0(seq(ntaxa(ps.noncontam)))
ps.noncontam
remove(dna)
```

```{r}
ps.noncontam = readRDS("~/BRAVE_Kids/psBRAVE.RDS")

#Subset to NP only samples and remove Indeterminate COVID postiive/negative results
ps.NP = subset_samples(ps.noncontam, sample_data(ps.noncontam)$method_np == "NP" & sample_data(ps.noncontam)$corona != "Indeterminate") %>%
  prune_species(speciesSums(.) > 0, .) 

ps.NP.r = transform_sample_counts(ps.NP,function(x) x/sum(x)) 

saveRDS(ps.NP, "~/BRAVE_Kids/psNP.RDS")
```

```{r, include = FALSE}
# Function to make matrix usable for vegan from phyloseq object
vegan_otu <- function(physeq) {
    OTU <- otu_table(physeq)
    if (taxa_are_rows(OTU)) {
        OTU <- t(OTU)
    }
    return(as(OTU, "matrix"))
}
```

#Code for Figures 1A and 1B
```{r}
sampleDF = sampleDF %>%
    mutate(agebin = case_when(age < 2 ~ '0-2',
                              age < 5 ~ '3-5',
                              age < 8 ~ '6-8',
                              age < 11 ~ '9-11',
                              age < 14 ~ '12-14',
                              age < 17 ~ '15-17',
                              TRUE ~ '18-20'))

Shannon = estimate_richness(ps.NP, measures = "Shannon") %>%
  cbind(., sampleDF, by = "names") 

Shannon$agebin = factor(Shannon$agebin, levels = c('0-2','3-5','6-8','9-11','12-14','15-17','18-20'))

sampleDF = sampleDF %>%
    mutate(Group1 = case_when(Group == 'Negative' ~ 'Negative',
                              Group == 'PosResp' ~ 'Positve (respiratory SX)',
                              Group == 'Pos_NoResp' ~ 'Positive (no respiratory SX)'))

ggplot(Shannon, aes(x=agebin, y=Shannon)) +
  geom_point(aes(color=Group1), alpha = 0.3) +
  scale_color_manual(values=c("darkslategray4", "mediumorchid4","#BF812D")) +
  stat_summary(aes(y=Shannon, group = Group1, color = Group1), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Age (years)")+
  ylab("Shannon diversity") + 
  labs(color = "SARS-CoV-2 status") +
  scale_y_continuous(limits = c(0,4)) + 
  theme(legend.position = "none")

Chao = estimate_richness(ps.NP, measures = "Chao1") %>%
  cbind(., sampleDF, by = "names") 

Chao$agebin = factor(Chao$agebin, levels = c('0-2','3-5','6-8','9-11','12-14','15-17','18-20'))

ggplot(Chao, aes(x=agebin, y=Chao1)) +
  geom_point(aes(color=Group1), alpha = 0.3) +
  scale_color_manual(values=c("darkslategray4", "mediumorchid4","#BF812D")) +
  stat_summary(aes(y=Chao1, group = Group1, color = Group1), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Age (years)")+
  ylab("Chao1 richness") + 
  labs(color = "SARS-CoV-2 status") +
  scale_y_continuous(limits = c(0,250)) + 
  theme(legend.position = "none")

#4 by 5
```

#Code for Figure 1C
```{r}
ord = ordinate(ps.NP.r, method = "NMDS", distance = "bray", trymax = 1000)

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

sampleDF$agebin = factor(sampleDF$agebin, levels = c('0-2','3-5','6-8','9-11','12-14','15-17','18-20'))

sampleDF = sampleDF %>%
    mutate(facet = case_when(agebin == '0-2' ~ '0-2 years',
                              agebin == '3-5' ~ '3-5 years',
                              agebin == '6-8' ~ '6-8 years',
                              agebin == '9-11' ~ '9-11 years',
                              agebin == '12-14' ~ '12-14 years',
                              agebin == '15-17' ~ '15-17 years',
                              TRUE ~ '18-20 years'))

sampleDF$facet = factor(sampleDF$facet, levels = c('0-2 years','3-5 years','6-8 years','9-11 years','12-14 years','15-17 years','18-20 years'))

sampleDF = sampleDF %>%
    mutate(Group1 = case_when(Group == 'Negative' ~ 'Negative',
                              Group == 'PosResp' ~ 'Positve (respiratory symptoms)',
                              Group == 'Pos_NoResp' ~ 'Positive (no respiratory symptoms)'))

sample_data(ps.NP)$Group1 = sampleDF$Group1

ordDF = cbind(sampleDF, ord$points)
ordDF2=ordDF
ordDF$agebin=NULL
ordDF$facet=NULL

p=ggplot(data = ordDF, aes(x = MDS1, y = MDS2)) +
  geom_point(data=ordDF, alpha=0.05) + 
  geom_point(data = ordDF2, aes(color = Group1), alpha=0.7) + 
  facet_wrap(~ facet, ncol=4) +
  scale_color_manual(values = c("darkslategray4", "mediumorchid4","#BF812D")) + 
  theme_bw() +
  xlab("NMDS1") + 
  ylab("NMDS2") +
  labs(color = "SARS-CoV-2 status")+ 
  scale_y_continuous(labels = scales::number_format(accuracy = 0.1))

#5.5 by 12
```

#Code to generate Figure 2
```{r}
#create age bins and lots of long code to make mean abundance profiles

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

sampleDF = sampleDF %>%
    mutate(agebin = case_when(age < 3 ~ '0-2',
                              age < 6 ~ '3-5',
                              age < 9 ~ '6-8',
                              age < 12 ~ '9-11',
                              age < 15 ~ '12-14',
                              age < 18 ~ '15-17',
                              TRUE ~ '18-20'))

sample_data(ps.NP.r)$agebin = sampleDF$agebin

ps.Genus = tax_glom(ps.NP.r, taxrank = "genus") 

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '0-2')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF1 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '3-5')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF2 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '6-8')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF3 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '9-11')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF4 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '12-14')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF5 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '15-17')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF6 = genusDF$C1

psC1 = subset_samples(ps.Genus, sample_data(ps.Genus)$agebin == '18-20')

genusDF = as.data.frame(cbind(t(otu_table(psC1, taxa_are_rows = FALSE)), tax_table(psC1)))

genusDF$domain = NULL
genusDF$phylum=NULL
genusDF$class=NULL
genusDF$order=NULL
genusDF$family=NULL
rownames(genusDF) = genusDF$genus
genusDF$genus=NULL
genusDF$species = NULL
genusDF = mutate_all(genusDF, function(x) as.numeric(as.character(x)))

genusDF$C1 = rowMeans(genusDF)

AvgDF7 = genusDF$C1 

AvgDF = cbind(AvgDF1, AvgDF2, AvgDF3, AvgDF4, AvgDF5, AvgDF6, AvgDF7) %>%
  data.frame(.)

rownames(AvgDF) = rownames(genusDF)

AvgDF$Avg = rowMeans(AvgDF)

fig = top_n(AvgDF, 9, Avg) 

Other = 1 - colSums(fig)

fig = rbind(fig, Other)

fig$tax = rownames(fig)

fig[10,9] = "Other"

fig = select(fig, -Avg) %>%
  dplyr::rename('0-2' = AvgDF1,
                '3-5' = AvgDF2,
                '6-8' = AvgDF3,
                '9-11' = AvgDF4,
                '12-14' = AvgDF5,
                '15-17' = AvgDF6,
                '18-20' = AvgDF7)

library(reshape2)
f = melt(fig) 
library(RColorBrewer)

mycolors = c("#E41A1C","#4A72A6","#48A462","#7E6E85","#EC83BA","#FFB716","#e6550d","#B75F49","#5ab4ac","#666666")

f$tax = factor(f$tax, levels=c("Anaerococcus","Corynebacterium","Dolosigranulum","Fusobacterium","Lawsonella","Moraxella", "Peptoniphilus", "Staphylococcus","Streptococcus","Other"))


ggplot(f, aes(fill = tax, y=value, x=variable)) +
  geom_bar(position="stack", stat="identity", color = "black", size = 0.25) +
  scale_fill_manual(values = mycolors, labels = c(expression(italic("Anaerococcus")),expression(italic("Corynebacterium")),expression(italic("Dolosigranulum")),expression(italic("Fusobacterium")),expression(italic("Lawsonella")),expression(italic("Moraxella")), expression(italic("Peptoniphilus")), expression(italic("Staphylococcus")),expression(italic("Streptococcus")),"Other")) +
  xlab("Age (years)") +
  ylab("Mean relative abundance") +
  labs(fill = "Genera") +
  theme_minimal() +
  scale_y_continuous(limits = c(0,1), expand = c(0, 0))+
theme(legend.text.align = 0)
```

#ANCOM by corona
```{r}
meta_data <- data.frame(sample_data(ps.NP))
meta_data$sampleID = rownames(meta_data)
otu_data <- data.frame(otu_table(ps.NP))
otu_id <- row.names(otu_data)
otu_data <- data.frame(t(otu_table(ps.NP)))
colnames(otu_data) <- otu_id

feature_table = otu_data

sample_var = "sampleID" 
group_var = NULL
out_cut = 0.05 
zero_cut = 0.95 
lib_cut = 1000 
neg_lb = TRUE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, out_cut, zero_cut, lib_cut, neg_lb)

feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

main_var = "corona"
p_adj_method = "BH"
alpha = 0.05
adj_formula = "age" 
rand_formula = NULL
res_cor = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, alpha, adj_formula, rand_formula)

res = res_cor$out %>%
  subset(., detected_0.8 == TRUE) %>%
  rename(ASV = taxa_id)

write.csv(res, "~/BRAVE_Kids/ancom2_corona_0.95_final.csv")

rm(adj_formula, alpha, group_var, lib_cut, main_var, neg_lb, otu_id, out_cut, p_adj_method, rand_formula, sample_var, struc_zero, zero_cut, feature_table, meta_data, otu_data, prepro)
```

#Figure 3A
```{r}
ancom_CorPN = read.csv("~/BRAVE_Kids/ancom2_corona_0.95_final.csv")

ancom_CorPN$ASV = as.character(ancom_CorPN$ASV)

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

corDF = sampleDF %>%
  select(., corona)

t=as.data.frame(ps.NP.r@tax_table@.Data[ancom_CorPN$ASV,]) %>%
  select(., 'genus')

d=otu_table(ps.NP.r)[,ancom_CorPN$ASV] %>%
  t(.) %>%
  as.data.frame(.) 
  
d1=cbind(d, t) %>%
  mutate(., var = paste(genus, " ", "(ASV", " ", rownames(.), ")", sep = "")) %>%
  select(., -genus) 
  
rownames(d1) = NULL

d=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., corDF, by = 0) %>%
  select(., -Row.names) %>%
  melt(., id.vars = "corona")

d$value1 = d$value + 1E-6

ggplot(d, aes(x = corona, y = value1, fill = corona)) +
  geom_boxplot(show.legend = F) +
  expand_limits(y=0)  +
  scale_fill_manual(values = c("darkslategray4", "cornflowerblue"))+
  theme_bw() +
  facet_wrap(~ variable, ncol=4) +
  ylab("Relative abundance") +
  xlab("SARS-CoV-2 status") +
  scale_y_continuous(breaks=c(0.0000001, 0.000001, 0.000001, 0.0001, 0.01, 1), trans = 'log') 
```

#ANCOM by respiratory symptoms
```{r}
ps.NP.pos = subset_samples(ps.NP,sample_data(ps.NP)$corona != "Negative")

meta_data <- data.frame(sample_data(ps.NP.pos))
meta_data$sampleID = rownames(meta_data)
otu_data <- data.frame(otu_table(ps.NP.pos))
otu_id <- row.names(otu_data)
otu_data <- data.frame(t(otu_table(ps.NP.pos)))
colnames(otu_data) <- otu_id

feature_table = otu_data

sample_var = "sampleID" 
group_var = "resp_sx_any"
out_cut = 0.05 
zero_cut = 0.95 
lib_cut = 1000 
neg_lb = TRUE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, out_cut, zero_cut, lib_cut, neg_lb)

feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

main_var = "resp_sx_any"
p_adj_method = "BH"
alpha = 0.05
adj_formula = "age" 
rand_formula = NULL
res_cor = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, alpha, adj_formula, rand_formula)

res = res_cor$out %>%
  subset(., detected_0.8 == TRUE) %>%
  rename(ASV = taxa_id) %>%
  subset(., W!="Inf")

write.csv(res, "~/BRAVE_Kids/ancom2_resp_0.95_final.csv")

rm(adj_formula, alpha, group_var, lib_cut, main_var, neg_lb, otu_id, out_cut, p_adj_method, rand_formula, sample_var, struc_zero, zero_cut, feature_table, meta_data, otu_data, prepro)
```

#Figure 3B
```{r}
ancom_CorResp = read.csv("~/BRAVE_Kids/ancom2_resp_0.95_final.csv")

ancom_CorResp$ASV = as.character(ancom_CorResp$ASV)

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

sampleDF = sampleDF %>%
    mutate(PosBin = case_when(Group == 'Negative' ~ 'Negative',
                              Group == 'PosResp' ~ 'Yes',
                              Group == 'Pos_NoResp' ~ 'No'))

RespDF = sampleDF %>%
  select(., PosBin)

ps.Pos.r = subset_samples(ps.NP.r, sample_data(ps.NP.r)$corona == "Positive")

t=as.data.frame(ps.Pos.r@tax_table@.Data[ancom_CorResp$ASV,]) %>%
  select(., 'genus')

d=otu_table(ps.Pos.r)[,ancom_CorResp$ASV] %>%
  t(.) %>%
  as.data.frame(.) 
  
d1=cbind(d, t) %>%
  mutate(., var = paste(genus, " ", "(ASV", " ", rownames(.), ")", sep = "")) %>%
  select(., -genus) 
  
rownames(d1) = NULL

d=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., RespDF, by = 0) %>%
  select(., -Row.names) %>%
  melt(., id.vars = "PosBin")

d$value1 = d$value + 1E-6

ggplot(d, aes(x = PosBin, y = value1, fill = PosBin)) +
  geom_boxplot(show.legend = F) +
  expand_limits(y=0)  +
  scale_fill_manual(values = c("mediumorchid4","#BF812D"))+
  theme_bw() +
  facet_wrap(~ variable, ncol=3, scales = 'free_y') +
  ylab("Relative abundance") +
  xlab("Respiratory symptoms") +
  scale_y_continuous(breaks=c(0.0000001, 0.000001, 0.000001, 0.0001, 0.01, 1), trans = 'log') 
```

#ANCOM by age
```{r}
meta_data <- data.frame(sample_data(ps.NP))
meta_data$sampleID = rownames(meta_data)
otu_data <- data.frame(otu_table(ps.NP))
otu_id <- row.names(otu_data)
otu_data <- data.frame(t(otu_table(ps.NP)))
colnames(otu_data) <- otu_id

feature_table = otu_data

sample_var = "sampleID" 
group_var = "age_cat"
out_cut = 0.05 
zero_cut = 0.95 
lib_cut = 1000 
neg_lb = TRUE
prepro = feature_table_pre_process(feature_table, meta_data, sample_var, group_var, out_cut, zero_cut, lib_cut, neg_lb)

feature_table = prepro$feature_table # Preprocessed feature table
meta_data = prepro$meta_data # Preprocessed metadata
struc_zero = prepro$structure_zeros # Structural zero info

main_var = "age"
p_adj_method = "BH"
alpha = 0.05
adj_formula = "Group" 
rand_formula = NULL
res_cor = ANCOM(feature_table, meta_data, struc_zero, main_var, p_adj_method, alpha, adj_formula, rand_formula)

res = res_cor$out %>%
  subset(., detected_0.8 == TRUE) %>%
  rename(ASV = taxa_id) %>%
  subset(., W!="Inf")

write.csv(res, "~/BRAVE_Kids/ancom2_age_0.95_final.csv")

rm(adj_formula, alpha, group_var, lib_cut, main_var, neg_lb, otu_id, out_cut, p_adj_method, rand_formula, sample_var, struc_zero, zero_cut, feature_table, meta_data, otu_data, prepro)
```

#Figure 4A
```{r}
ancom_age = read.csv("~/BRAVE_Kids/ancom2_age_0.95_final.csv")

#Same ASVs present, only showing those 9
ancomR = ancom_age[ancom_CorResp$ASV,]

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

sampleDF = sampleDF %>%
    mutate(agebin = case_when(age < 2 ~ '0-2',
                              age < 5 ~ '3-5',
                              age < 8 ~ '6-8',
                              age < 11 ~ '9-11',
                              age < 14 ~ '12-14',
                              age < 17 ~ '15-17',
                              TRUE ~ '18-20'))

sampleDF$agebin = factor(sampleDF$agebin, levels = c('0-2','3-5','6-8','9-11','12-14','15-17','18-20'))

sampleDF = sampleDF %>%
    mutate(PosBin = case_when(GroupBin == 'Negative' ~ 'Negative',
                              GroupBin == 'PosResp' ~ 'Yes',
                              GroupBin == 'Pos_NoResp' ~ 'No'))

RespDF = sampleDF %>%
  select(., agebin, PosBin)

t=as.data.frame(ps.NP.r@tax_table@.Data[ancom_CorResp$ASV,]) %>%
  select(., 'genus')

d=otu_table(ps.NP.r)[,ancom_CorResp$ASV] %>%
  t(.) %>%
  as.data.frame(.) 
  
d1=cbind(d, t) %>%
  mutate(., var = paste(genus, " ", "(ASV", " ", rownames(.), ")", sep = "")) %>%
  select(., -genus) 
  
rownames(d1) = NULL

d=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., RespDF, by = 0) %>%
  select(., -Row.names, PosBin) %>%
  melt(., id.vars = "agebin")

d2=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., RespDF, by = 0) %>%
  select(., -Row.names, agebin) %>%
  melt(., id.vars = "PosBin")

d3 = merge(d, d2, by= 0) %>%
  select(., agebin, PosBin, value.x, variable.x)

d3$value.x = as.numeric(d3$value.x)

d3$value1 = d3$value.x + 1E-6

d3 = subset(d3, d3$variable.x != "PosBin" & d3$PosBin != "Negative")

p=ggplot(d3, aes(x=agebin, y=value1)) +
  geom_boxplot(aes(fill=PosBin)) +
  scale_fill_manual(values=c("mediumorchid4","#BF812D"), name = "Respiratory symptoms") +
  #stat_summary(aes(y=value1, group = GroupBin, color = GroupBin), fun.y = mean, geom = "line")+
  theme_bw()+
  xlab("Age (years)")+
  ylab("Relative abundance") + 
  facet_wrap(~ variable.x, ncol=3, scales = 'free_y') +
  scale_y_continuous(breaks=c(0.0000001, 0.000001, 0.000001, 0.0001, 0.01, 1), trans = 'log') 
```

#Figure 4B
```{r}
ancom_CorPN = read.csv("~/BRAVE_Kids/ancom2_corona_0.95_final.csv")

ancom_CorPN$ASV = as.character(ancom_CorPN$ASV)

sampleDF = sample_data(ps.NP.r) %>%
  data.frame(.)

sampleDF = sampleDF %>%
    mutate(agebin = case_when(age < 2 ~ '0-2',
                              age < 5 ~ '3-5',
                              age < 8 ~ '6-8',
                              age < 11 ~ '9-11',
                              age < 14 ~ '12-14',
                              age < 17 ~ '15-17',
                              TRUE ~ '18-20'))

sampleDF$agebin = factor(sampleDF$agebin, levels = c('0-2','3-5','6-8','9-11','12-14','15-17','18-20'))

corDF = sampleDF %>%
  select(., corona, agebin)

t=as.data.frame(ps.NP.r@tax_table@.Data[ancom_CorPN$ASV,]) %>%
  select(., 'genus')

d=otu_table(ps.NP.r)[,ancom_CorPN$ASV] %>%
  t(.) %>%
  as.data.frame(.) 
  
d1=cbind(d, t) %>%
  mutate(., var = paste(genus, " ", "(ASV", " ", rownames(.), ")", sep = "")) %>%
  select(., -genus) 
  
rownames(d1) = NULL

d=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., corDF, by = 0) %>%
  select(., -Row.names, agebin) %>%
  melt(., id.vars = "agebin")

d2=tibble::column_to_rownames(d1,var = "var") %>%
  t(.) %>%
  as.data.frame(.) %>%
  merge(., corDF, by = 0) %>%
  select(., -Row.names) %>%
  melt(., id.vars = "corona")

d3 = merge(d, d2, by= 0) %>%
  select(., agebin, corona, value.x, variable.x) %>%
  subset(., variable.x != "corona")

d3$value.x = as.numeric(d3$value.x)

d3$value1 = d3$value.x + 1E-6

ggplot(d3, aes(x = agebin, y = value1, fill = corona)) +
  geom_boxplot() +
  expand_limits(y=0)  +
  scale_fill_manual(values = c("darkslategray4", "cornflowerblue"), name = "SARS-CoV-2 status")+
  theme_bw() +
  facet_wrap(~ variable.x) +
  ylab("Relative abundance") +
  xlab("Age (years)") +
  scale_y_continuous(breaks=c(0.0000001, 0.000001, 0.000001, 0.0001, 0.01, 1), trans = 'log') 
```
